napari_animation/_tests/test_animation.py

```diff
@@ -139,3 +144,41 @@ def test_end_state_reached(image_animation):
     image_animation.capture_keyframe(steps=2)
     last_state = image_animation._frames[-1]
     assert last_state == image_animation.key_frames[-1].viewer_state
+
+
+@pytest.mark.parametrize("layer_class, data, ndim", layer_test_data)
+def test_attributes_for_all_layer_types(
+    make_napari_viewer, layer_class, data, ndim
+):
+    """Test that attributes are in the viewer_state for all napari layer types"""
+    viewer = make_napari_viewer()
+    add_layer_by_type(viewer, layer_class, data, visible=True)
+    layer_animation = Animation(viewer)
+    # get the state of the layer
+    layer_state = viewer.layers[0]._get_state()
+    # remove attributes that arn't captured
+    layer_state.pop("metadata")
+    layer_state.pop("data", None)
```

Comment:
Can we drop these lines now, for the same reason as the surface hack? Or maybe do something like:

```
original_layer_state = {
    k: v for k, v in layer_state.items()
    if k in CAPTURED_LAYER_ATTRIBUTES
}
```

or something like this? I don't like that this stuff is "hardcoded" in the test...

Reply:
(also, `arn't` is a typo. 😜)

Reply:
Metadata isn't captured
https://github.com/napari/napari-animation/blob/35e5accac9c6994b9735bc7ecb1d679e2b70bdc5/napari_animation/viewer_state.py#L17-L19
and it's popped in viewer_state
https://github.com/napari/napari-animation/blob/35e5accac9c6994b9735bc7ecb1d679e2b70bdc5/napari_animation/viewer_state.py#L34
So I think that one has to be popped.
`data` doesn't get captured either:
https://github.com/napari/napari-animation/blob/35e5accac9c6994b9735bc7ecb1d679e2b70bdc5/napari_animation/viewer_state.py#L30C43-L31
Only the 2 element of the tuple is used.
While it could be cool if it worked, I'm not sure how keyframes and interpolation would work with data.

So the idea in the test would be to make a NEVER_CAPTURED_ATTR variable at the top of the file and put those there?

Reply:
> While it could be cool if it worked, I'm not sure how keyframes and interpolation would work with data.

I a user wants to use napari-animation to change smoothly change the contents of `data` through out, there's a very good chance they actually need to add a dimension to `Dims` and use that instead :P